doc/luasnip.txt

```diff
@@ -3628,197 +3673,423 @@ GENERAL ~
+
+Return whether the current node is inside a choiceNode.
+
+
+change_choice(val)                                 *luasnip-api-change_choice*
+
+Change the currently active choice.
+
+- `val: (1|-1)` Move one choice forward or backward.
+
+
+set_choice(choice_indx)                               *luasnip-api-set_choice*
+
+Set the currently active choice.
+
+- `choice_indx: integer` Index of the choice to switch to.
+
+
+get_current_choices(): string[]              *luasnip-api-get_current_choices*
+
+Get a string-representation of all the current choiceNode’s choices. This
+function returns:
+
+- `string[]` -concatenated lines of every choice.
+
+
+active_update_dependents()              *luasnip-api-active_update_dependents*
+
+Update all nodes that depend on the currently-active node.
+
+
+store_snippet_docstrings(snippet_table) *luasnip-api-store_snippet_docstrings*
+
+Generate and store the docstrings for a list of snippets as generated by
+`get_snippets()`. The docstrings are stored at
+`stdpath("cache") .. "/luasnip/docstrings.json"`, are indexed by their trigger,
+and should be updated once any snippet changes.
+
+- `snippet_table: { [string]: LuaSnip.Snippet[] }` A table mapping some keys to lists of snippets
+    (keys are most likely filetypes).
+
+
+load_snippet_docstrings(snippet_table)   *luasnip-api-load_snippet_docstrings*
+
+Provide all passed snippets with a previously-stored (via
+`store_snippet_docstrings`) docstring. This prevents a somewhat costly
+computation which is performed whenever a snippets’ docstring is first
+retrieved, but may cause larger delays when `snippet_table` contains many of
+snippets. Utilize this function by calling
+`ls.store_snippet_docstrings(ls.get_snippets())` whenever snippets are
+modified, and `ls.load_snippet_docstrings(ls.get_snippets())` on startup.
+
+- `snippet_table: { [string]: LuaSnip.Snippet[] }` List of snippets, should contain the same keys
+    (filetypes) as the table that was passed to `store_snippet_docstrings`. Again, most likely the
+    result of `get_snippets`.
+
+
+unlink_current_if_deleted()            *luasnip-api-unlink_current_if_deleted*
+
+Checks whether (part of) the current snippet’s text was deleted, and removes
+it from the jumplist if it was (it cannot be jumped back into).
+
+
+exit_out_of_region(node)                      *luasnip-api-exit_out_of_region*
+
+Checks whether the cursor is still within the range of the root-snippet `node`
+belongs to. If yes, no change occurs; if no, the root-snippet is exited and its
+`i(0)` will be the new active node. If a jump causes an error (happens mostly
+because the text of a snippet was deleted), the snippet is removed from the
+jumplist and the current node set to the end/beginning of the next/previous
+snippet.
+
+- `node: LuaSnip.Node`
+
+
+filetype_extend(ft, extend_ft)                   *luasnip-api-filetype_extend*
+
+Add `extend_ft` filetype to inherit its snippets from `ft`. Example:
```

Comment:
🤔 This `Example` should be on its own line as well, this one is surprising though, because I know I left a blank line before it in the Lua doc annotation:
![image](https://github.com/user-attachments/assets/3db2e015-0f49-46ee-8f22-456ee5d75a3d)

But the markdown is all mashed together (will be an issue as we add more top documentation to functions & stuff!):
![image](https://github.com/user-attachments/assets/360769ea-6777-4e71-b9c1-11aefd3ff1b2)

And what's going on with the ``  `sh`/`zsh`/`bash` `zsh`/`bash` `bash` buffers. `` at the end 👀
Did your md generator invent some stuff here? (is it sentient??? 😱)

Reply:
Again, this needed a few larger changes, 68faade forces an empty line between two paragraphs, 2354788 replaces the very naive previous algorithm for keeping something like `` (`sh`) `` together with something that is more robust.
So, it's not about to become Skynet, I just appended all `[^%s]`-chars to code-spans, which includes other code-spans xD

Reply:
Oh, and totally agree, full empty lines are paragraph-breaks, they should be preserved :+1:

Reply:
Thanks for saving the world this time 🙏😄

Yay for newlines 🚀

Reply:
Didn't even have to time travel :muscle: